### ISSUE TITLE
Issue that Google Chrome, Edge browsers stop playing videos when the window is in Shrink/PiP mode

### ISSUE THREAD

**Issue by gileli121** [BiGilSoft/WindowTop#146]
More users complain about issue #117 with google chrome.
Google has a built-in feature that is designed to disable video rendering when it thinks that it is invisible.
This is in order to save computer hardware resources usage.

As a result, you get this issue when you Shrink window:
![image](https://user-images.githubusercontent.com/17680514/134407185-171c8f4b-81f8-4d30-88d5-af1ac47c5b49.png)

(Screenshot from #117)

The issue is that chrome thinks by mistake that it is invisible when it is actually not.
In Shrink + Live Preview mode, the real window is indeed invisible.
However, the window is cloned using Windows DWM API to another window owned by WindowTop

In this case, chrome is not invisible. The only invisible thing is the chrome window.
But you still see the window via the Shrink mode (a second window that cloning it).

So what we need to do to handle this issue:

**Option 1** 
Open bug for and report about it so it will be fixed in google chrome browser
https://chromium.googlesource.com/chromium/src 

**Option 2**
The software will suggest the user to open the following link:
chrome://flags/#calculate-native-win-occlusion 

When you put the link in chrome, it will take you to here:
![image](https://user-images.githubusercontent.com/17680514/133322691-26a0aeba-a9cf-404d-b5ab-50cf19312d3c.png)

The software will suggest the user disable this feature:
![image](https://user-images.githubusercontent.com/17680514/133322767-6b944f61-4236-498d-8389-430f5259135e.png)

Once the feature is disabled, we will no longer get issue #117 

## Update:
In the end, I did not use Option 1 and not Option 2.
I did Option 3. That is tricking chrome.
**Option 3**
See from 
https://github.com/BiGilSoft/WindowTop/issues/146#issuecomment-924130293

**Comment by gileli121** [comment]
Bug opened for chromium-based browsers (Google Chrome, and the new Microsoft Edge)
https://bugs.chromium.org/p/chromium/issues/detail?id=1249583

**Comment by gileli121** [comment]
While I reviewed the source code of google chrome (the chromium source code), I found that I can trick it here:
https://github.com/chromium/chromium/blob/ef3c0b7e3f9387e57570cdfd6c7e65ee5add4ec9/ui/aura/native_window_occlusion_tracker_win.cc#L744

The hack:
```cpp
HWND taskListThumbnailWnd = FindWindow(L"TaskListThumbnailWnd", NULL);
ShowWindow(taskListThumbnailWnd, SW_SHOW);
```

The following code will trick chrome and will cause it to think that it should disable calculating the occlusion state and it will assume that any window is visible and as a result, it will turn on rendering back for all windows.


I will implement this hack so WindowTop will do it for any chromium based browser (That are Chrome and Edge)

**Comment by gileli121** [comment]
Build with the hotfix
[WindowTop Build 22-09-21_1.zip](https://github.com/BiGilSoft/WindowTop/files/7206349/WindowTop.Build.22-09-21_1.zip)
:

This hotfix was implemented for the following chromium-based browsers:
```c#
  case "chrome.exe":
  case "msedge.exe":
  case "brave.exe":
  case "vivaldi.exe":
  case "epic.exe":
  case "slimbrowser.exe":
  case "dragon.exe":
  case "AvastBrowser.exe":
```

**Comment by gileli121** [comment]
This hack is not working right in Windows 10.
It working fine in Windows 11.

In Windows 10, there is a problem with the taskbar preview
![image](https://user-images.githubusercontent.com/17680514/134313661-0369e37c-c91e-48f9-899c-d8c4857ee531.png)

**Comment by gileli121** [comment]
The following trick working without side effects

```cpp
WNDCLASS wc = {};

wc.lpfnWndProc = DefWindowProc;
wc.hInstance = GetModuleHandle(NULL);
wc.lpszClassName = L"TaskListThumbnailWnd";
auto success = RegisterClass(&wc);

HWND fakeTLThumb = CreateWindowEx(
WS_EX_TOOLWINDOW,  
L"TaskListThumbnailWnd",         
L"TaskListThumbnailWnd_Fake",  
WS_POPUP,
-9999, -9999, 0, 0,
NULL,
NULL,
GetModuleHandle(NULL),
NULL
);


HWND orgTLThumb = FindWindow(L"TaskListThumbnailWnd", NULL);

bool orgTLThumbIsVisible = !IsWindowVisible(orgTLThumb);
bool orgTLThumbIsVisible_new = !orgTLThumbIsVisible;

while (true)
{
orgTLThumbIsVisible_new = IsWindowVisible(orgTLThumb);
if (orgTLThumbIsVisible_new != orgTLThumbIsVisible)
{
	orgTLThumbIsVisible = orgTLThumbIsVisible_new;

	if (!orgTLThumbIsVisible)
	{
		ShowWindow(fakeTLThumb, SW_SHOWNOACTIVATE);
	}
	else
	{
		ShowWindow(fakeTLThumb, SW_HIDE);
	}
}

Sleep(1000);
}
```

I will change the implementation

**Comment by gileli121** [comment]
It passed the tests in Windows 10 and Windows 11

**Comment by gileli121** [comment]
New build for testing:

[WindowTop Build 14-09-22_1.zip](https://github.com/BiGilSoft/WindowTop/files/7210489/WindowTop.Build.14-09-22_1.zip)

**Comment by gileli121** [comment]
The last version of the hack did not work perfectly after more testing.

Sometimes the video will be blank in the preview.

**Comment by gileli121** [comment]
Released in v5.5.8
